flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ContextResolvedTableJsonSerializer.java

```diff
@@ -0,0 +1,98 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.serde;
+
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.api.config.TableConfigOptions;
+import org.apache.flink.table.api.config.TableConfigOptions.CatalogPlanCompilation;
+import org.apache.flink.table.catalog.ContextResolvedTable;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.JsonGenerator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.SerializerProvider;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+
+/** JSON serializer for {@link ContextResolvedTable}. */
+class ContextResolvedTableJsonSerializer extends StdSerializer<ContextResolvedTable> {
+    private static final long serialVersionUID = 1L;
+
+    public static final String FIELD_NAME_IDENTIFIER = "identifier";
+    public static final String FIELD_NAME_CATALOG_TABLE = "catalogTable";
+
+    public ContextResolvedTableJsonSerializer() {
+        super(ContextResolvedTable.class);
+    }
+
+    @Override
+    public void serialize(
+            ContextResolvedTable contextResolvedTable,
+            JsonGenerator jsonGenerator,
+            SerializerProvider serializerProvider)
+            throws IOException {
+        final CatalogPlanCompilation planCompilationOption =
+                SerdeContext.get(serializerProvider)
+                        .getConfiguration()
+                        .get(TableConfigOptions.PLAN_COMPILE_CATALOG_OBJECTS);
+
+        if (contextResolvedTable.isAnonymous()
+                && planCompilationOption == CatalogPlanCompilation.IDENTIFIER) {
+            throw cannotSerializeAnonymousTable(contextResolvedTable.getIdentifier());
+        }
+
+        jsonGenerator.writeStartObject();
+
+        if (!contextResolvedTable.isAnonymous()) {
+            // Serialize object identifier
+            jsonGenerator.writeObjectField(
+                    FIELD_NAME_IDENTIFIER, contextResolvedTable.getIdentifier());
+        }
+
+        if ((contextResolvedTable.isPermanent() || contextResolvedTable.isAnonymous())
+                && planCompilationOption != CatalogPlanCompilation.IDENTIFIER) {
+            // Pass to the ResolvedCatalogTableJsonSerializer the option to serialize or not the
+            // identifier
+            serializerProvider.setAttribute(
```

Comment:
I really find it weird that a serializer sets attributes.

Reply:
It is, I removed it